Changes to episodes 1 and 2

catavallejos opened the pull request:
Changes as agreed in HackMD.

Notes:

1) I simplified the proposed overview provided at the end of episode 1. This is to avoid introducing too many context that are not yet explained. Instead, I tried to focus on concrete biological examples, rather than discussing some of the technical details (e.g. robustness of clusters).

2) Reference for the methylation data is TBC.

Alanocallaghan commented on _episodes_rmd/02-high-dimensional-regression.Rmd:
````diff
@@ -102,13 +103,10 @@ knitr::kable(head(colData(methylation)), row.names = FALSE)
+methylation. The following heatmap summarises age and methylation levels 
+available in the Prostate dataset:
 
-```{r heatmap, echo=FALSE, fig.cap="Visualising the data as a heatmap, it's clear that there's too many models to fit 'by hand'.", fig.alt="Heatmap of methylation values across all features. Samples are ordered according to age."}
````
if you're gonna show this heatmap code I'd change it to use pheatmap or similar, the ComplexHeatmap is fairly complex and has a lot of options, will be likely to distract from the stats

catavallejos replied:
Good point. We decided to include the code because the students were asking for it during delivery. I will try to simplify.

catavallejos replied:
Not quite sure the change to `pheatmap` is necessary. Here are the potential edits I suggest:

1) Continue using `ComplexHeatmap` but simplify some of the parameters (e.g. do not specify a custom colour-scale). The code and output would be as follows:

```
library("ComplexHeatmap")
order <- order(age)
age_ord <- age[order]
methyl_mat_ord <- methyl_mat[, order]

Heatmap(methyl_mat_ord,
        name = "M-value",
        cluster_columns = FALSE,
        show_row_names = FALSE,
        show_column_names = FALSE,
        row_title = "Feature",
        column_title =  "Sample",
        top_annotation = columnAnnotation(age = age_ord))
```

![image](https://user-images.githubusercontent.com/7511093/216772389-721bb8ad-d1ce-43f2-84a9-638bfcc800cd.png)

2) Edit to use `pheatmap`. The code and output would be as follows (I still need to figure out how to add the "Feature" and "Sample" axis labels, but this gives an idea of the required code). 

```
library("pheatmap")
order <- order(age)
age_ord <- age[order]
age_ord_df <- data.frame(Age = age_ord)
rownames(age_ord_df) <- colnames(methyl_mat_ord)
methyl_mat_ord <- methyl_mat[, order]
pheatmap(methyl_mat_ord,
         annotation_col = age_ord_df,
         cluster_cols = FALSE, 
         cluster_rows = TRUE,
         show_colnames = FALSE, 
         show_rownames = FALSE))
```
![image](https://user-images.githubusercontent.com/7511093/216772485-a43457b7-7b91-44a4-83f5-8dadcaed4792.png)


I am inclined to go for option 1), but open to suggestions

catavallejos replied:
Either way, we can add a couple of code comments to briefly explain some of the parameters.

Alanocallaghan replied:
Yeah 1 seems fine, the colorscale was just in case it wasn't centered on zero (so then white/neutral is like 0.5) but not the case here.

catavallejos replied:
great! changing to option 1.

Alanocallaghan commented on _episodes_rmd/02-high-dimensional-regression.Rmd:
```diff
@@ -1090,10 +1122,11 @@ experiment over and over.
--   [limma user
+-   [**`limma`** user
     manual](https://www.bioconductor.org/packages/release/bioc/vignettes/limma/inst/doc/usersguide.pdf).
+-   [The **`VariancaPartition`** package](https://bioconductor.org/packages/release/bioc/vignettes/variancePartition/inst/doc/dream.html) has similar functionality than **`limma`** but allows the inclusion of random effects. 
```
```suggestion
-   [The **`VariancePartition`** package](https://bioconductor.org/packages/release/bioc/vignettes/variancePartition/inst/doc/dream.html) has similar functionality than **`limma`** but allows the inclusion of random effects. 
```

Alanocallaghan commented:
If you want general refs for the methylation data you'll probably find them here: https://github.com/immunomethylomics/FlowSorted.Blood.EPIC

catavallejos commented:
Thanks for the comments @Alanocallaghan !

> If you want general refs for the methylation data you'll probably find them here: https://github.com/immunomethylomics/FlowSorted.Blood.EPIC

More than a general reference, we were looking for the exact source from which the methylation data was downloaded. Any chance you have that?

Alanocallaghan commented:
All the data download scripts are here, with the same name as the original files, I don't think any are too arcane but do let me know if any should be clearer. I can add some details to the readme or in the course materials maybe? https://github.com/carpentries-incubator/high-dimensional-stats-r/tree/main/data

I found the download link here: https://bioconductor.org/packages/release/data/experiment/vignettes/FlowSorted.Blood.EPIC/inst/doc/FlowSorted.Blood.EPIC.html

Alanocallaghan commented:
On the topic of data, I think there's maybe some unused files. eg `data/small_methylation.rds`

catavallejos commented:
Thanks @Alanocallaghan.

I edited the episode to put a link to the download code (which I updated to have a link to the download instructions).

catavallejos commented:
I also deleted `small_methylation.R` and `small_methylation.Rds` as they are currently unused (and, if needed, we can always subset `methylation` as it was done [here](https://github.com/carpentries-incubator/high-dimensional-stats-r/blob/main/data/small_methylation.R)

catavallejos commented:
PS: I will leave the other datasets for now, just in case they are useful for future course developments

catavallejos commented:
@hannesbecher that's all my edits pushed. Unless you have further comments, please merge whenever you have the time.

catavallejos commented:
one minor extra edit: for consistency, renamed exercises as challenges and added numbers.

@hannesbecher could you please check the PCA etc episodes to make sure it's consistent there too?

hannesbecher commented:
> one minor extra edit: for consistency, renamed exercises as challenges and added numbers.
> 
> @hannesbecher could you please check the PCA etc episodes to make sure it's consistent there too?

Going to check this!

hannesbecher commented on _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:
````diff
@@ -1,387 +1,398 @@
+
+```{r table-intro, echo = FALSE}
+knitr::include_graphics("../fig/intro-table.png")
+# ![Figure 1: Example of high-dimensional biomedical data](D:/Statistical consultancy/Consultancy/Grant applications/UKRI teaching grant 2021/Working materials/Table for Intro.png)
````
Could removed this comment

catavallejos replied:
done!

hannesbecher commented on _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:
````diff
@@ -1,387 +1,397 @@
+> >
+> > pairs(Prostate)  #plot each pair of variables against each other
+> > ```
+> > The `pairs` function plots relationships between each of the variables in
````
I suggest adding parentheses to `pairs` to make it consistent with other functions (even though you say that it is a function).

catavallejos replied:
done

hannesbecher commented on _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:
````diff
@@ -1,387 +1,397 @@
+> > pairs(Prostate)  #plot each pair of variables against each other
+> > ```
+> > The `pairs` function plots relationships between each of the variables in
+> > the `prostate` dataset. This is possible for datasets with smaller numbers
````
Should `Prostate` have a capital P?

hannesbecher commented on _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:
```diff
@@ -1,387 +1,397 @@
+
+> ## Challenge 3
+> 
+> Use the cor() function to examine correlations between all variables in the 
```
I suggest to add backticks around cor() to make is show as code.

catavallejos replied:
done

hannesbecher commented on _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:
````diff
@@ -1,387 +1,397 @@
+> > ```
+> > 
+> > As seen above, some variables are highly correlated. In particular, the 
+> > correlation between gleason and pgg45 is equal to 0.75.
````
I'd also suggest backticks around the feature names `gleason` and `pgg45`.

hannesbecher commented on _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:
````diff
@@ -1,387 +1,397 @@
+> > summary(model3)
+> > ```
+> >
+> > Although gleason and pgg45 have statistically significant univariate effects,
````
Same here.

hannesbecher commented on _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:
```diff
@@ -1,387 +1,397 @@
+variables, (2) regularised regression, (3) dimensionality reduction, and (4) 
+clustering. Here are some examples for when each of these approaches may be used:
+
+(1) Regression with numerous outcome refers to situations in which there are 
```
outcome -> outcomes
or add the word variables

catavallejos replied:
done

hannesbecher commented on _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:
```diff
@@ -1,387 +1,397 @@
+molecular data, a typical example are *differential gene expression* analyses. 
+We will explore this type of analysis in the *Regression with many outcomes* episode.
+
+(2) Regularisation (also known as regularised regression or penalised regression) 
```
Not sure what you think - could make *regularised regression* and *penalised regression* show in italics?

catavallejos replied:
done - added italics to highlight new concepts

hannesbecher commented on _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:
```diff
@@ -1,387 +1,397 @@
+
+(3) Dimensionality reduction is commonly used on high dimensional datasets for 
+data exploration or as a preprocessing step prior to other downstream analyses. 
+For instance, a low dimensional visualisation of a gene expression dataset may
```
I would suggest adding a hyphen: low-dimensional. (But not feeling to strong about this.)

catavallejos replied:
done (this keeps consistency!)

catavallejos commented on _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:
````diff
@@ -1,387 +1,397 @@
+> > summary(model2)
+> > ```
+> >
+> > Based on these results we conclude that both gleason and pgg45 have a 
````
done

hannesbecher commented on _episodes_rmd/02-high-dimensional-regression.Rmd:
```diff
@@ -102,46 +106,41 @@ knitr::kable(head(colData(methylation)), row.names = FALSE)
-interest (age). It is clear from the heatmap that there are too many
-features to do so manually.
+1. To predict age using methylation leves as predictors. In this case, we would 
+have a single outcome (age) which will be predicted using 5000 covariates 
```
Could have backticks around `age`

hannesbecher commented on _episodes_rmd/02-high-dimensional-regression.Rmd:
```diff
@@ -102,46 +106,41 @@ knitr::kable(head(colData(methylation)), row.names = FALSE)
-> ## Exercise
+2. To predict methylation levels using age as a predictor. In this case, we 
+would have 5000 outcomes (methylation levels across the genome) and a single 
+covariate (age). 
```
here too

hannesbecher commented on _episodes_rmd/02-high-dimensional-regression.Rmd:
````diff
@@ -305,6 +305,7 @@ from the model object:
 
 ```{r fit1}
 age <- methylation$Age
+# methyl_mat[1, ] indicates that the 1st CpG will be used as outcome variable
````
Well done.

hannesbecher reviewed (approved):
Looks good to me. Thank you!
